### PR TITLE
DEV: Add a skeleton for section landing page & items

### PR DESCRIPTION
**Pull request by martin-brennan**
We are going to start making section landing pages
for admin for each sidebar section. This lays the framework
with routes and simple components that can be further
refined by a designer, but I have taken the base CSS from
AI which Kris made.

The initial section landing items will be used in AI to replace
the placeholders added in this commit https://github.com/discourse/discourse-ai/pull/813/commits/b8b3c614515b95ad642223a0ac24a37e6976b6bd

**Icon & Link, Mobile/Desktop** - This is the default section landing look

image
image

**Subtitle with Button, Mobile/Desktop** - This is what AI will use

image
image

### CODE REVIEW

File: app/assets/javascripts/admin/addon/templates/section-account.hbs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @descriptionLabel="admin.section_landing_pages.account.backups.description"
+    @titleRoute="admin.backups"
+  />
+  <AdminSectionLandingItem
+    @icon="gift"
+    @titleLabel="admin.section_landing_pages.account.whats_new.title"
+    @descriptionLabel="admin.section_landing_pages.account.whats_new.description"
+    @titleRoute="admin.whatsNew"
+  />
+  <AdminSectionLandingItem
```
Comment (martin-brennan) [inline app/assets/javascripts/admin/addon/templates/section-account.hbs:23]: Will remove this at a later PR, just for testing purposes ATM

File: app/assets/javascripts/admin/addon/components/admin-section-landing-item.gjs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  PrimaryButton,
+} from "admin/components/admin-page-action-button";
+
+export default class AdminSectionLandingItem extends Component {
+  get title() {
+    if (this.args.titleLabelTranslated) {
+      return this.args.titleLabelTranslated;
+    } else if (this.args.titleLabel) {
+      return i18n(this.args.titleLabel);
+    }
```
Comment (lis2) [inline app/assets/javascripts/admin/addon/components/admin-section-landing-item.gjs:18]:
```suggestion
    return this.args.titleLabelTranslated || (this.args.titleLabel && i18n(this.args.titleLabelTranslated))
```

File: app/assets/javascripts/admin/addon/components/admin-section-landing-item.gjs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        <div class="admin-section-landing-item__icon">
+          {{dIcon @icon}}
+        </div>
+      {{/if}}
+      <div class="admin-section-landing-item__content">
+        {{#if this.subTitle}}
+          <h4
+            class="admin-section-landing-item__subtitle"
+          >{{this.subTitle}}</h4>
+        {{/if}}
```
Comment (lis2) [inline app/assets/javascripts/admin/addon/components/admin-section-landing-item.gjs:60]: It feels a bit weird that the subtitle is above the title. What would you say about naming it "keyword" or something similar?
Or is it a mistake and subtitle should be below?

File: app/controllers/admin/section_controller.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class Admin::SectionController < Admin::AdminController
+  def show
+  end
```
Comment (lis2) [inline app/controllers/admin/section_controller.rb:5]: I think that this empty method can be deleted. It should work as long as this path is defined in routes

**Review by lis2 — APPROVED** [review]

File: app/assets/javascripts/admin/addon/components/admin-section-landing-item.gjs (hunk 3 above)
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/components/admin-section-landing-item.gjs:60]: It's correct that it should be above, but yeah I agree subtitle is a bit of a weird name for it :thinking: Apparently it's called a tagline, will use that https://www.w3.org/TR/2014/REC-html5-20141028/common-idioms.html#sub-head https://ux.stackexchange.com/questions/103390/name-of-small-text-above-title

File: app/controllers/admin/section_controller.rb (hunk 4 above)
Comment (martin-brennan) [reply app/controllers/admin/section_controller.rb:5]: I wasn't able to remove this, rails complained

Comment (lis2) [reply app/controllers/admin/section_controller.rb:5]: Thank you for checking and sorry for bad idea :)

Comment (martin-brennan) [reply app/controllers/admin/section_controller.rb:5]: No probs! It sounded reasonable :)

File: app/assets/javascripts/admin/addon/routes/admin-section-test.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import DiscourseRoute from "discourse/routes/discourse";
+
+export default class AdminSectionTest extends DiscourseRoute {}
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/addon/routes/admin-section-test.js:3]: do we need this file?

Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/routes/admin-section-test.js:3]: Sorry no, will kill it

File: app/assets/javascripts/admin/addon/components/admin-section-landing-item.gjs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+  get subTitle() {
+    if (this.args.subTitleLabelTranslated) {
+      return this.args.subTitleLabelTranslated;
+    } else if (this.args.subTitleLabel) {
+      return i18n(this.args.subTitleLabel);
+    }
+  }
+
+  get cssClasses() {
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/components/admin-section-landing-item.gjs:37]: I thought I would do something more complex here but I didn't, will switch to concactClass

**Pull request merged**